src/AppInstallerCLICore/Workflows/UninstallFlow.cpp

```diff
@@ -0,0 +1,201 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "pch.h"
+#include "UninstallFlow.h"
+#include "WorkflowBase.h"
+#include "MSStoreInstallerHandler.h"
+#include "ShellExecuteInstallerHandler.h"
+#include "AppInstallerMsixInfo.h"
+
+using namespace AppInstaller::Manifest;
+using namespace AppInstaller::Msix;
+using namespace AppInstaller::Repository;
+
+namespace AppInstaller::CLI::Workflow
+{
+    namespace
+    {
+        std::optional<DWORD> InvokeCreateProcess(const std::string& command, IProgressCallback& progress)
+        {
+            std::wstring commandUtf16 = Utility::ConvertToUTF16(command);
+
+            // Parse the command string as application and command line for CreateProcess
+            PWSTR app, commandLine;
+            THROW_IF_FAILED(SHEvaluateSystemCommandTemplate(commandUtf16.c_str(), &app, &commandLine, NULL));
+
+            STARTUPINFOW startupInfo = { sizeof(startupInfo) };
+            PROCESS_INFORMATION processInfo;
+            BOOL success = CreateProcessW(app, commandLine, NULL, NULL, FALSE, 0, NULL, NULL, &startupInfo, &processInfo);
+
+            CoTaskMemFree(app);
```

Comment:
>CoTaskMemFree [](start = 12, length = 13)

If these require free'ing, you should use the wil RAII classes.

Reply:
Thanks. I didn't know about that.